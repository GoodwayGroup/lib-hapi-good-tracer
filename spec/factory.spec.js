const nock = require('nock');
const axiosFactory = require('../lib/axiosFactory');

describe('axiosFactory', () => {
    it('return an axios client', () => {
        const client = axiosFactory({ headers: { common: { 'user-agent': 'test-mctestings' } } });
        expect(client.interceptors.request).toBeDefined();
        expect(client.interceptors.response).toBeDefined();
        expect(client.defaults.headers.common).toEqual({
            Accept: 'application/json, text/plain, */*',
            'user-agent': 'test-mctestings'
        });
    });

    it('passthrough a good response', async () => {
        const client = axiosFactory({ headers: { common: { 'user-agent': 'test-mctestings' } } });

        nock('http://test.com')
            .get('/test')
            .reply(200, 'success');

        const ret = await client.get('http://test.com/test');
        expect(ret.status).toBe(200);
        expect(ret.data).toBe('success');
    });

    it('generate boom error on bad repsonse', async (done) => {
        const client = axiosFactory({ headers: { common: { 'user-agent': 'test-mctestings' } } });

        nock('http://test.com')
            .get('/test')
            .reply(404, 'nope');

        try {
            await client.get('http://test.com/test');
            done.fail();
        } catch (e) {
            expect(e.message).toBe('Request failed with status code 404');
            expect(e).toMatchSnapshot();
            done();
        }
    });

    it('generate boom error on bad repsonse [DEV_EXPOSE_AXIOS_ERRORS enabled]', async (done) => {
        process.env.DEV_EXPOSE_AXIOS_ERRORS = true;
        const client = axiosFactory();

        nock('http://test.com')
            .get('/test')
            .reply(404, 'nope');

        try {
            await client.get('http://test.com/test');
            done.fail();
        } catch (e) {
            expect(e.message).toBe('Request failed with status code 404');
            expect(e).toMatchSnapshot();
            done();
        }
        delete process.env.DEV_EXPOSE_AXIOS_ERRORS;
    });
});
