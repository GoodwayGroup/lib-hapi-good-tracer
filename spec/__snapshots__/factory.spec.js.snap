// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axiosFactory generate boom error on bad repsonse [DEV_EXPOSE_AXIOS_ERRORS enabled] 1`] = `[Error: Request failed with status code 404]`;

exports[`axiosFactory generate boom error on bad repsonse 1`] = `[Error: Request failed with status code 404]`;
